### PR TITLE
Fix PackTool nuspec has p2p dependencies

### PR DESCRIPTION
**Pull request by wli3**
Fix https://github.com/dotnet/cli/issues/8724

### CODE REVIEW

File: src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs [hunk 1]
```diff
@@ -63,5 +63,31 @@ public void It_contains_dependencies_dll()
                 }
             }
         }
+
+        [Fact]
+        public void It_remove_other_package_dependency()
```
Comment (livarcocc) [inline src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs]: It_removes_other_package_dependencies

**Comment by livarcocc** [comment]
Can you give some background with what is the issue and fix here?

**Review by nguerrera — APPROVED** [review]

**Comment by wli3** [comment]
@livarcocc if you `dotnet pack` a p2p project. nuget by default will convert the Library project to a package reference and it ends up in nuspec. Something like packageId = "library", version = 1.0.0.

If we don't remove this project then become package reference. It will fail to install due to "cannot find the package -- packageId = "library", version = 1.0.0"

**Comment by nguerrera** [comment]
Was there a bug filed for this?

File: src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs (hunk 1 above)
Comment (livarcocc) [reply src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs]: Given your explanation, how about naming this It_does_not_add_p2p_references_as_package_references_to_nuspec?

**Comment by wli3** [comment]
@nguerrera no, I find it by manual tests. Let me create one

File: src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs (hunk 1 above)
Comment (nguerrera) [reply src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs]: I like that second suggestion better too.

Comment (wli3) [reply src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProjectWithP2PReference.cs]: Renamed as the second one

**Pull request merged**